Spanner/src/Session/CacheSessionPool.php

```diff
@@ -313,6 +314,9 @@ public function acquire($context = SessionPoolInterface::CONTEXT_READ)
 
         // If we don't have a session, let's wait for one or throw an exception.
         if (!$session) {
+            if (!is_null($exception)) {
+                throw $exception;
```

Comment:
Pls add UT for this case.

Reply:
Done! Thank you @vishwarajanand for the review!